rust-src/wallet_library/src/identity.rs

```diff
@@ -116,88 +84,31 @@ pub fn create_id_request_with_keys_v1_aux(input: IdRequestInputWithKeys) -> Resu
     Ok(to_string(&response)?)
 }
 
-/// Creates an identity object request where the supplied seed phrase is
-/// used to derive the required keys.
-pub fn create_id_request_with_seed_v1_aux(input: IdRequestInputWithSeed) -> Result<JsonString> {
-    let seed_decoded = hex::decode(&input.seed_as_hex)?;
-    let seed: [u8; 64] = match seed_decoded.try_into() {
-        Ok(s) => s,
-        Err(_) => bail!("The provided seed {} was not 64 bytes", input.seed_as_hex),
-    };
-
-    let wallet: ConcordiumHdWallet = ConcordiumHdWallet {
-        seed,
-        net: input.net,
-    };
-
-    let identity_provider_index = input.common.ip_info.ip_identity.0;
-    let identity_index = input.identity_index;
-    let prf_key: prf::SecretKey<ArCurve> =
-        wallet.get_prf_key(identity_provider_index, identity_index)?;
-    let id_cred_sec: PedersenValue<ArCurve> =
-        PedersenValue::new(wallet.get_id_cred_sec(identity_provider_index, identity_index)?);
-    let blinding_randomness: concordium_base::id::ps_sig::SigRetrievalRandomness<
-        constants::IpPairing,
-    > = wallet.get_blinding_randomness(identity_provider_index, identity_index)?;
-
-    let input = IdRequestInputWithKeys {
-        common: input.common,
-        prf_key,
-        id_cred_sec,
-        blinding_randomness: base16_encode_string(&blinding_randomness),
-    };
-
-    create_id_request_with_keys_v1_aux(input)
-}
-
-/// The shared identity recovery request input that is independent
-/// of whether or not the request is going to be created
-/// by supplying a seed phrase or the secret key material
-/// directly.
+/// Required input for generating an identity recovery request.
 #[derive(SerdeSerialize, SerdeDeserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct IdRecoveryRequestCommonInput {
+pub struct IdentityRecoveryRequestInput {
     ip_info:        IpInfo<constants::IpPairing>,
     global_context: GlobalContext<constants::ArCurve>,
     timestamp:      u64,
-}
-
-/// Required input for generating an identity recovery request. This input
-/// contains the private key which will be used directly to generate
-/// the request.
-#[derive(SerdeSerialize, SerdeDeserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct IdRecoveryRequestInputWithKeys {
-    common:      IdRecoveryRequestCommonInput,
-    id_cred_sec: PedersenValue<ArCurve>,
-}
-
-/// Required input for generating an identity recovery request from a seed. The
-/// required private key will be derived from the seed.
-#[derive(SerdeSerialize, SerdeDeserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct IdRecoveryRequestInputWithSeed {
-    common:         IdRecoveryRequestCommonInput,
-    seed_as_hex:    String,
-    net:            Net,
-    identity_index: u32,
+    id_cred_sec:    PedersenValue<ArCurve>,
 }
 
 #[derive(SerdeSerialize, SerdeDeserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct IdRecoveryRequestOut {
+pub struct IdentityRecoveryRequestOut {
```

Comment:
If I wanted to be nitpicky, I would suggest to call this `Output` with analogy to `Input`. :)